Lib/fontTools/varLib/cff.py

```diff
@@ -365,24 +378,24 @@ def merge_charstrings(glyphOrder, num_masters, top_dicts, masterModel):
 
 		# If the charstring required a new model, create
 		# a VarData table to go with, and set vsindex.
+		key = tuple(v is not None for v in all_cs)
 		try:
-			key = tuple(v is not None for v in all_cs)
 			vsindex = vsindex_by_key[key]
 		except KeyError:
-			varTupleIndexes = []
-			for support in model.supports[1:]:
-				if support not in masterSupports:
-					masterSupports.append(support)
-				varTupleIndexes.append(masterSupports.index(support))
-			var_data = varLib.builder.buildVarData(varTupleIndexes, None, False)
-			vsindex = len(vsindex_dict)
-			vsindex_by_key[key] = vsindex
-			vsindex_dict[vsindex] = (model, [key])
-			varDataList.append(var_data)
+			vsindex = _add_new_vsindex(model, key,masterSupports, vsindex_dict,
+				vsindex_by_key, varDataList)
 		# We do not need to check for an existing new_cs.private.vsindex,
 		# as we know it doesn't exist yet.
 		if vsindex != 0:
 			new_cs.program[:0] = [vsindex, 'vsindex']
+			
+	# If there is no variation in any of the charstrings, then vsindex_dict
+	# never gets built. This is could still be needed if there is variation 
+	# in the PrivatDict, so we will build the default data for vsindex = 0.
+	if not vsindex_dict:
+		key = (True)*num_masters
```

Comment:
I think you meant a tuple here:

```suggestion
		key = (True,) * num_masters
```

otherwise the value of `(True) * num_masters` is equivalent to multiplying boolean `True` (which is the same as integer `1` in python) by another integer (`num_masters`).